pages/blog/_posts/banger.mdx

```diff
@@ -0,0 +1,279 @@
+---
+heading: A Deep Dive into a Video Rendering Pipeline
+showSubtitle: true
+subtitle: banger.show is a video app maker that heavily relies on background data processing
+image: /assets/blog/banger/banger-main.jpg
+date: 2024-05-01
+author: Igor Samokhovets
+disableCTA: true
+---
+
+import YouTube from 'react-youtube-embed';
+
+Hi everyone! My name is [Igor Samokhovets](https://twitter.com/IgorSamokhovets) and I'm a music producer who goes by the artist name Tequila Funk. Together with Mark Beziaev we created [banger.show](https://banger.show/) -- a video maker app for musicians, DJs, and labels. It allows music industry people to create stunning visual assets for their music.
+
+<YouTube id="eSLKm6IfZM4" height={300}/>
+
+ Creating a video for your new song takes only a few minutes and you don't need to know any complex software because banger.show works in your browser.
+
+
+## Making background processing snappy
+
+At banger.show, we do a lot of background processing. Managing render states, generating visual assets in the background, and even handling the rendering process on remote distributed workers. We chose Inngest because there's no better way to handle background jobs if you're using Next.js.
+
+Let's dive into some parts of the video rendering pipeline.
+
+### 1. Updating the render status
+
+The first step in the render pipeline is to update the render status and set user credits on hold
+
+```js
+const updatedVideo = await step.run('Update user balance', async () => {
+  await dbConnect()
+
+  const render = await VideoModel.findOneAndUpdate(
+    { _id: videoId },
+    { $set: { renderProgress: 0, renderTime: 0, status: 'pending' } },
+    { new: true }
+  )
+    .populate('user')
+    .lean()
+
+  invariant(video, 'no render found')
+
+  // Simplified
+  await UserModel.updateOne(
+    {
+      _id: video.user._id
+    },
+    {
+      $inc: { unitsRemaining: -video.videoDuration }
+    }
+  )
+
+  return video
+})
+```
+
+### 2. Cropping the audio file
+
+In banger.show, user should select a fragment of the song to create a videoization for it. In the background job, we:
```

Comment:
```suggestion
In banger.show, the user selects a fragment of the song to create a "videoization" for it. In the background job, we:
```
Added quotes b/c I would guess videoization is their own amalgamation of video and visualization.